readthedocs/restapi/views/integrations.py

```diff
@@ -64,6 +76,11 @@ def post(self, request, project_slug):
                 return Response(resp, status=status.HTTP_406_NOT_ACCEPTABLE)
         except Project.DoesNotExist:
             raise NotFound('Project not found')
+        if not self.is_payload_valid():
+            return Response(
+                {'detail': self.invalid_payload_msg},
+                status=HTTP_400_BAD_REQUEST
+            )
```

Comment:
I wonder if we should log these with a warning. It feels like something that shouldn't happen often, and if it does fail, we should likely know about it.

Reply:
I'd also like to `log.warning` this.